src/System.Management.Automation/CoreCLR/CorePsPlatform.cs

```diff
@@ -882,21 +883,37 @@ public static CommonStat GetStat(string path)
             public static int GetProcFSParentPid(int pid)
             {
                 const int invalidPid = -1;
-
-                // read /proc/<pid>/stat
-                // 4th column will contain the ppid, 92 in the example below
-                // ex: 93 (bash) S 92 93 2 4294967295 ...
-                var path = $"/proc/{pid}/stat";
+                TimeSpan matchTimeout = new(0, 0, 5);  // 5 Seconds
+
+                // read /proc/<pid>/status
+                // Row beginning with PPid: \d is the parent process id.
+                // This used to check /proc/<pid>/stat but that file was meant
+                // to be a space delimited line but it contains a value which
+                // could contain spaces itself. Using the status file is a lot
+                // simpler because each line contains a record with a simple
+                // label.
+                // https://github.com/PowerShell/PowerShell/issues/17541#issuecomment-1159911577
+                var path = $"/proc/{pid}/status";
                 try
                 {
-                    var stat = System.IO.File.ReadAllText(path);
-                    var parts = stat.Split(' ', 5);
-                    if (parts.Length < 5)
+                    using FileStream fs = File.OpenRead(path);
+                    using StreamReader sr = new(fs);
+                    string line;
+                    while ((line = sr.ReadLine()) != null)
                     {
-                        return invalidPid;
+                        Match ppidMatch = Regex.Match(
+                            line,
+                            @"^PPid:\s+(\d+)$",
```

Comment:
The proposal looks very expensive. I'd prefer #12925.

Reply:
Is the concern around the regex, the reading of multiple lines or something else? If it's the regex this could be changes to just doing `line.StartsWith("PPid: ")` and extract it from there but I thought regex gives some nicer validation.

My only concern about #12925 is if any extra entries are added to the `stat` file in the future that could mess with the `LastIndexOf(')')` check. I have no idea what the chances are of that happening but the file certainly has had additions in the past so I can't rule it out.

Reply:
@jborean93 is there any concern using `StartsWith()` and then splitting out the pid trimming the whitespace instead of a regex?

Reply:
The regex just ensure it's `PPid: digit` but considering this is probably mandated in the spec it's probably not something to be concerned about. I can change it to `StartsWith("PPid: ")` but didn't think regex would be that bad of a performance problem.

Reply:
@jborean93 Any chance to get the proposed changes done? Thanks!

Reply:
Sorry will get to this next week, just on holidays at the moment.

Reply:
Has been updated to use `StartsWith` instead as per the suggestion.